solidity/test/isms/OptimismISM.t.sol

```diff
@@ -0,0 +1,461 @@
+
+    /* ============ hook.postDispatch ============ */
+
+    function testPostDispatch() public {
+        deployAll();
+
+        vm.selectFork(mainnetFork);
+
+        bytes memory encodedMessage = _encodeTestMessage(
+            0,
+            address(testRecipient)
+        );
+        bytes32 messageId = Message.id(encodedMessage);
+
+        bytes memory encodedHookData = abi.encodeCall(
+            OptimismISM.verifyMessageId,
+            (address(this), messageId)
+        );
+
+        uint40 nonce = ICanonicalTransactionChain(L1_CANNONICAL_CHAIN)
+            .getQueueLength();
+
+        vm.expectEmit(true, true, true, false, L1_MESSENGER_ADDRESS);
+        emit SentMessage(
+            address(opISM),
+            address(opHook),
+            encodedHookData,
+            nonce,
+            DEFAULT_GAS_LIMIT
+        );
+
+        opHook.postDispatch(OPTIMISM_DOMAIN, messageId);
+    }
+
+    function testPostDispatch_ChainIDNotSupported() public {
+        deployAll();
+
+        vm.selectFork(mainnetFork);
+
+        bytes32 messageId = Message.id(
+            _encodeTestMessage(0, address(testRecipient))
+        );
+
+        vm.expectRevert("OptimismHook: invalid destination domain");
+        opHook.postDispatch(11, messageId);
+    }
+
+    /* ============ ISM.verifyMessageId ============ */
+
+    function testverifyMessageId() public {
+        deployAll();
+
+        vm.selectFork(optimismFork);
+
+        bytes32 messageId = Message.id(
+            _encodeTestMessage(0, address(testRecipient))
+        );
+
+        bytes memory encodedHookData = abi.encodeCall(
+            OptimismISM.verifyMessageId,
+            (address(this), messageId)
+        );
+
+        (uint240 nonce, uint16 verison) = Encoding.decodeVersionedNonce(
+            l2Messenger.messageNonce()
+        );
+        uint256 versionedNonce = Encoding.encodeVersionedNonce(
+            nonce + 1,
+            verison
+        );
+
+        bytes32 versionedHash = Hashing.hashCrossDomainMessageV1(
+            versionedNonce,
+            address(opHook),
+            address(opISM),
+            0,
+            DEFAULT_GAS_LIMIT,
+            encodedHookData
+        );
+
+        vm.startPrank(
+            AddressAliasHelper.applyL1ToL2Alias(L1_MESSENGER_ADDRESS)
+        );
+
+        vm.expectEmit(true, true, false, false, address(opISM));
+        emit ReceivedMessage(address(this), messageId);
+
+        vm.expectEmit(true, false, false, false, L2_MESSENGER_ADDRESS);
+        emit RelayedMessage(versionedHash);
+
+        l2Messenger.relayMessage(
+            versionedNonce,
+            address(opHook),
+            address(opISM),
+            0,
+            DEFAULT_GAS_LIMIT,
+            encodedHookData
+        );
+
+        assertEq(opISM.verifiedMessageIds(messageId, address(this)), true);
+
+        vm.stopPrank();
+    }
+
+    // function testverifyMessageId_WithValue() public {
```

Comment:
file a ticket or link to something like https://github.com/hyperlane-xyz/issues/issues/355 inline please